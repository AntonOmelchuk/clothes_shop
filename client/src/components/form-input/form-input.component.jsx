import React from 'react';

import {
  FormInputContainer,
  FormInputLabel,
  GroupContainer,
} from './form-input.style';

export const FormInput = ({label, handleChange, ...restPros}) => (
  <GroupContainer>
    <FormInputContainer
      className='form-input'
      onChange={handleChange}
      {...restPros}
    />
    {label ? (
      <FormInputLabel className={`${restPros.value.length ? 'shrink' : ''}`}>
        {label}
      </FormInputLabel>
    ) : null}
  </GroupContainer>
);
