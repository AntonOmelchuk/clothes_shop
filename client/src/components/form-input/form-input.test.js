import {shallow} from 'enzyme';
import React from 'react';

import {FormInput} from './form-input.component';

let wrapper;
let mockHandleChange;

beforeEach(() => {
  mockHandleChange = jest.fn();

  const mockProps = {
    label: 'email',
    value: 'test@gmail.com',
    handleChange: mockHandleChange,
  };

  wrapper = shallow(<FormInput {...mockProps} />);
});

it('renders FormInput component', () => {
  expect(wrapper).toMatchSnapshot();
});
