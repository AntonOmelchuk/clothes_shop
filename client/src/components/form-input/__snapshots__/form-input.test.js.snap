// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormInput component 1`] = `ShallowWrapper {}`;
